### PR TITLE
Allow overriding configmap images

### PR DESCRIPTION
**Pull request by crandles**
Allow overriding all images the operator uses and adds additional "PullPolicy" flags for configmap images.

### CODE REVIEW

**Comment by crandles** [comment]
I'm not sure its necessary to add the pullPolicy flags, but I wanted to allow the same fields to be modified for all images in the values.yaml

**Comment by mgoodness** [comment]
I have a few comments on the Helm chart-related changes, but will hold off until @brancz has a chance to weigh in.

**Comment by brancz** [comment]
I'm not sure I understand why this would be useful aside from debugging. The tag is bumped whenever anything changes in the respective images, which are all controlled by us, so I don't see the necessity to adapt this configuration.

Generally the point of the Prometheus Operator is so one does not have to think about details like this, but I'm happy to be convinced with a good use case :slightly_smiling_face: .

**Comment by crandles** [comment]
I can drop the PullPolicy, I don't have a specific need for that. I wasn't sure about adding it in the first place.

What I need from the PR is the option to set the configmap images, so I can use my own internal docker hub hosted images.

**Comment by crandles** [comment]
Alternatively, those images could be configured via the AlertmanagerSpec/PrometheusSpec, instead of the current prometheus-operator's CLI options. Would that be a better solution for this?

**Comment by brancz** [comment]
The idea is that those images are merely utility, so we assume that they will be the same for all Pods, we didn't feel it was just as strong of an argument for the Prometheus/Alertmanager images.

In that sense as setting all images to custom repos is already possible, I'd suggest to revert the changes to the operator and just add it to the helm chart.

**Comment by brancz** [comment]
Now it's up to @mgoodness :slightly_smiling_face:

File: helm/prometheus-operator/README.md [hunk 1]
```diff
@@ -80,6 +80,10 @@ Parameter | Description | Default
 `global.hyperkube.repository` | Hyperkube image | `quay.io/coreos/hyperkube`
 `global.hyperkube.tag` | Hyperkube image tag | `v1.6.2_coreos.0`
 `global.hyperkube.pullPolicy` | Hyperkube image pull policy | `IfNotPresent`
+`global.prometheus-config-reloader.repository` | prometheus-config-reloader image | `quay.io/coreos/prometheus-config-reloader`
+`global.prometheus-config-reloader.tag` | prometheus-config-reloader tag | `v0.0.1`
+`global.configmap-reload.repository` | configmap-reload image | `quay.io/coreos/configmap-reload`
+`global.configmap-reload.tag` | configmap-reload tag | `v0.0.1`
 `image.repository` | Image | `quay.io/coreos/prometheus-operator`
```
Comment (mgoodness) [inline helm/prometheus-operator/README.md]: Let's "de-globalize" these. I'll be doing the same for the hyperkube values.

Also, please stick to camelCase.

File: helm/prometheus-operator/templates/deployment.yaml [hunk 2]
```diff
@@ -25,6 +25,8 @@ spec:
           {{- if .Values.kubeletService.enable }}
             - --kubelet-service={{ .Values.kubeletService.namespace }}/{{ .Values.kubeletService.name }}
           {{- end }}
+            - --prometheus-config-reloader={{ .Values.global.prometheus-config-reloader.repository }}:{{ .Values.global.prometheus-config-reloader.tag }}
+            - --config-reloader-image={{ .Values.global.configmap-reload.repository }}:{{ .Values.global.configmap-reload.tag }}
```
Comment (mgoodness) [inline helm/prometheus-operator/templates/deployment.yaml]: Drop `.global` and make camelCase.

File: helm/prometheus-operator/values.yaml [hunk 3]
```diff
@@ -6,6 +6,16 @@ global:
...
+  ## Prometheus-config-reloader image to use for config and rule reloading
+  prometheus-config-reloader:
+    repository: quay.io/coreos/prometheus-config-reloader
+    tag: v0.0.1
+
+  ## Configmap-reload image to use for reloading configmaps
+  configmap-reload:
+    repository: quay.io/coreos/configmap-reload
+    tag: v0.0.1
+
```
Comment (mgoodness) [inline helm/prometheus-operator/values.yaml]: Move out of `global` and make camelCase.

**Comment by mgoodness** [comment]
Pushed to opsgoodness repo.

**Pull request merged**